index-scheduler/src/lib.rs

```diff
@@ -242,47 +242,68 @@ impl IndexScheduler {
         self.index_mapper.indexes(&rtxn)
     }
 
-    /// Returns the tasks corresponding to the query.
-    pub fn get_tasks(&self, query: Query) -> Result<Vec<Task>> {
+    /// Return the task ids corresponding to the query
+    pub fn get_task_ids(&self, query: &Query) -> Result<RoaringBitmap> {
         let rtxn = self.env.read_txn()?;
-        let last_task_id = match self.last_task_id(&rtxn)? {
-            Some(tid) => query.from.map(|from| from.min(tid)).unwrap_or(tid),
-            None => return Ok(Vec::new()),
-        };
 
         // This is the list of all the tasks.
-        let mut tasks = RoaringBitmap::from_sorted_iter(0..last_task_id).unwrap();
+        let mut tasks = {
+            let mut all_tasks = RoaringBitmap::new();
+            for status in [
+                Status::Enqueued,
+                Status::Processing,
+                Status::Succeeded,
+                Status::Failed,
+            ] {
+                all_tasks |= self.get_status(&rtxn, status)?;
+            }
+            all_tasks
+        };
```

Comment:
Do you think it's a good idea to create a new bitmap to store the whole available tasks? (in another PR)

Reply:
Here it looks like we are enumerating the whole set of variants of `Status` right? Shouldn't it be better to [use enum-iterator](https://lib.rs/crates/enum-iterator) or not?

Reply:
Yep why not, and we need it for the error message as well thus, it's not even a new dependency.